### PR TITLE
feat(eazy): Add support for setting password, username, url and custom messages

### PR DESCRIPTION
**Pull request by Uzaaft**
Just a simple PR with some functionality I added for my own usage. 

Feel free to close this if you think these are outside the scope of this library.

### CODE REVIEW

File: src/Easy.zig [hunk 1]
```diff
@@ -272,6 +272,22 @@ pub fn setWritefunction(
...
+}
+
+pub fn setURL(self: Self, url: [:0]const u8) !void {
+    try checkCode(c.curl_easy_setopt(self.handle, c.CURLOPT_URL, url.ptr));
+}
+
+pub fn setCustomMessageRequest(self: Self, message: [:0]const u8) !void {
+    try checkCode(c.curl_easy_setopt(self.handle, c.CURLOPT_CUSTOMREQUEST, message.ptr));
+}
+
```
Comment (jiacai2050) [inline src/Easy.zig:290]:
```suggestion
```

File: src/Easy.zig [hunk 2]
```diff
@@ -272,6 +272,14 @@ pub fn setWritefunction(
     try checkCode(c.curl_easy_setopt(self.handle, c.CURLOPT_WRITEFUNCTION, func));
 }
 
+pub fn setUser(self: Self, user: [:0]const u8) !void {
+    try checkCode(c.curl_easy_setopt(self.handle, c.CURLOPT_USERNAME, user.ptr));
```
Comment (jiacai2050) [inline src/Easy.zig:276]:
```suggestion
pub fn setUsername(self: Self, username: [:0]const u8) !void {
    try checkCode(c.curl_easy_setopt(self.handle, c.CURLOPT_USERNAME, username.ptr));
```

**Comment by jiacai2050** [comment]
Thanks.

I delete two methods, they are already defined with similar names.

**Comment by Uzaaft** [comment]
My bad, I didn't find those when I got myself familiar with the library.

**Comment by Uzaaft** [comment]
BTW, the PR title should've been fixed before merging. It's not factually correct anymore

**Pull request merged**